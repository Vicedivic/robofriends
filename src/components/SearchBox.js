import React from "react";

const SearchBox = ({ searchChange }) => {
  return (
    <div>
      <label htmlFor="search" style={{ fontSize: 0, position: "absolute" }}>Search</label>
      <input
        className="search"
        type="search"
        placeholder="search robots"
        onChange={searchChange}
        id="search"
      />
    </div>
  );
};

export default SearchBox;
