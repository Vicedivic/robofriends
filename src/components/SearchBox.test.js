import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import SearchBox from "./SearchBox";

describe("SearchBox", () => {
  const mockOnSearchChange = jest.fn();
  const renderer = new ShallowRenderer();
  const searchboxHtml = renderer.render(
    <SearchBox searchChange={mockOnSearchChange} />
  );

  it("renders properly", () => {
    expect(searchboxHtml).toMatchSnapshot();
  });
});
