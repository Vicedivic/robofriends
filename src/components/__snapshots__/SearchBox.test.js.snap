// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders properly 1`] = `
<div>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    onChange={[MockFunction]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
